deepspeed/runtime/engine.py

```diff
@@ -1144,6 +1163,30 @@ def _configure_distributed_model(self, model):
         if self.sequence_parallel_size > 1:
             self.communication_data_type = self._config.seq_parallel_communication_data_type
 
+        
+        if self.sequence_parallel_size > 1:
+            log_dist(f"Using DS sequence parallelism (SP) of size {self.sequence_parallel_size}", ranks=[0])
+            #replace module attention with deespeed dist attention
+            #Assert we have torch version with device_mesh for parallel group
+            #Assert flash attn#
+            assert pkg_version.parse(torch.__version__) >= pkg_version.parse("2.2.2"), \
+                f"HF model finetune with DeepSpeed SP  requires torch >= 2.2.2, you have version {torch.__version__}"
+            assert FLASH_ATTN_AVAILABLE, \
+                f"HF model finetune with DeepSpeed SP  requires Flash attention"
+
+            for _, module in self.module.named_modules():
+                if all(hasattr(module, attr) for attr in ['k_proj', 'v_proj', 'q_proj', 'out_proj']):
+                    attn = module
+                    compute_attn_sp = DistributedAttention(
+                        attn._flash_attention_forward,
```

Comment:
Flash attention implementation is not device agnostic and will not run all all platforms.
It's better to keep the current module forward and not replace it's implementation.

Reply:
Ulysses works best with flash attention. Example without flash attention is available in Megatron-DeepSpeed.